src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2972,7 +2972,7 @@ void os::pd_commit_memory_or_exit(char* addr, size_t size, bool exec,
   #define MADV_POPULATE_WRITE MADV_POPULATE_WRITE_value
 #else
   // Sanity-check our assumed default value if we build with a new enough libc.
-  static_assert(MADV_POPULATE_WRITE == MADV_POPULATE_WRITE_value);
+  static_assert(MADV_POPULATE_WRITE == MADV_POPULATE_WRITE_value, "MADV_POPULATE_WRITE != MADV_POPULATE_WRITE_value");
```

Comment:
Maybe use this instead?
```suggestion
  STATIC_ASSERT(MADV_POPULATE_WRITE == MADV_POPULATE_WRITE_value);
```

Reply:
This would look very strange, considering we have another very similar static_assert with message at  line 2985  just 10 lines below .

Reply:
I would then suggest that we clean that out as well. There's no need to keep adding these redundant error messages. Note that we started using STATIC_ASSERT and then later people started to add direct static_assert calls.